spec/system/settings/two_factor_authentication/recovery_codes_spec.rb

```diff
@@ -0,0 +1,28 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe 'Settings TwoFactorAuthentication RecoveryCodes' do
+  describe 'Generating recovery codes' do
+    let(:user) { Fabricate :user, otp_required_for_login: true }
+
+    before { sign_in(user) }
+
+    it 'updates the codes and includes them in the view' do
+      # Attempt to generate codes
+      visit settings_two_factor_authentication_methods_path
+      click_on I18n.t('two_factor_authentication.generate_recovery_codes')
+
+      # Fill in challenge password
+      fill_in 'form_challenge_current_password', with: user.password
+
+      expect { click_on I18n.t('challenge.confirm') }
+        .to(change { user.reload.otp_backup_codes })
+
+      expect(page)
+        .to have_content(I18n.t('two_factor_authentication.recovery_codes_regenerated'))
+        .and have_title(I18n.t('settings.two_factor_authentication'))
+        .and have_css('ol.recovery-codes')
```

Comment:
It is not testing for the inclusion of the codes themselves anymore. Is that intended?

Reply:
Oops!, good catch ... it was intentional to stop stubbing at the controller level, but I meant to preserve the check by moving the stub out to the recovery code generation level. Pushed update to change that.